### PR TITLE
task/initial-commit

### PR DESCRIPTION
**Pull request by rzareen**

### CODE REVIEW

File: EmailageJavaScriptWrapper/EmailValidator.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
+function emailValidate(requestObject, accountSId, authToken, baseUrl, requestType, dataType, sigMethod, nonce, version) {
+    if (baseUrl == undefined || (baseUrl.length) > 0 == false) {
```
Comment (martinkduncan) [inline EmailageJavaScriptWrapper/EmailValidator.js:2]: !baseUrl 
???
https://stackoverflow.com/questions/6003884/how-do-i-check-for-null-values-in-javascript#6003920

File: EmailageJavaScriptWrapper/EmailValidator.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    if (nonce)
+        authObject.oauth_nonce = nonce;
+
+    var oauthPostModel = getOAuthPostModel(requestType, authObject, baseUrl);
+    var oauthData = getOauthData(oauthPostModel);
+    var sig = oauthSign(authToken + '&', oauthData, sigMethod);
+    var requestUrl = oauthPostModel.action + "?" + getParameterString(oauthPostModel.parameters) + "&oauth_signature=" + percentEncode(sig);
+    var result = "";
+
+    $.ajax({
```
Comment (martinkduncan) [inline EmailageJavaScriptWrapper/EmailValidator.js:23]: jQuery required?

File: EmailageJavaScriptWrapper/EmailValidator.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    }).done(function (response) {
+        result = response;
+    });
+    return result;
+};
+
+var getAuthObject = function (dataType, accountSId, sigMethod, version) {
+
+    if (!sigMethod)
+        sigMethod = "HMAC-SHA1";
```
Comment (martinkduncan) [inline EmailageJavaScriptWrapper/EmailValidator.js:44]: sigMethod = "HMAC-SHA256"
???

File: README.md [hunk 4]
```diff
@@ -1 +1,11 @@
-# Emailage_Javascript
\ No newline at end of file
+# Emailage_Javascript
+This project contains javascript methods used to build the request object and submit an ajax request to the Emailage Email Validator. 
+
```
Comment (martinkduncan) [inline README.md:3]: # Dependencies
* jQuery
* jsSHA

**Comment by martinkduncan** [comment]
I like it!  nice and clean :-).  Do you want to put some tests in there too?

File: EmailageJavaScriptWrapper/EmailValidator.js (hunk 2 above)
Comment (rzareen) [reply EmailageJavaScriptWrapper/EmailValidator.js:23]: Using it just to build the sample form via the requestObj. The core functions are javascript.

File: EmailageJavaScriptWrapper/EmailValidator.js (hunk 1 above)
Comment (rzareen) [reply EmailageJavaScriptWrapper/EmailValidator.js:2]: you mean if(!baseUrl), right? There isn't a use for triple question mark. There is a triple !!!, but a single bang will do what I need here: https://stackoverflow.com/questions/19550294/what-does-the-syntax-mean-in-javascript (Sorry if I misunderstood and you were just saying,  "Zareen, come on now, clean that up." :)

File: EmailageJavaScriptWrapper/EmailValidator.js (hunk 3 above)
Comment (rzareen) [reply EmailageJavaScriptWrapper/EmailValidator.js:44]: Setting this as a default, honestly that can be HMAC-SHA256, HMAC-SHA234,HMAC-SHA512. I'll add a select drop down for this in the sample form

**Pull request merged**